File: Go/hello_go/loan_test.go
package main

import (
	"testing"
)

func TestCalculateLoanPrincipal(t *testing.T) {
	tests := []struct {
		A          float64
		years      int
		annualRate float64
		expected   int
	}{
		{50000, 10, 0.03, 5188120},
		{70000, 20, 0.02, 13860660},
		{50000, 10, 0.00001, 5999698},
	}

	for _, tt := range tests {
		result := calculateLoanPrincipal(tt.A, tt.years, tt.annualRate)
		if result != tt.expected {
			t.Errorf("calculateLoanPrincipal(%f, %d, %f) = %d; want %d", tt.A, tt.years, tt.annualRate, result, tt.expected)
		}
	}
}

File: Rust/hello_actix_web/tests/loan_test.rs
use hello_actix_web::loan::calculate_loan_principal;

#[test]
fn test_calculate_loan_principal() {
    let cases = [
        (50000.0, 10, 0.03, 5188120),
        (70000.0, 20, 0.02, 13860660),
        (50000.0, 10, 0.00001, 5999698),
    ];
    for (A, years, annual_rate, expected) in cases {
        assert_eq!(calculate_loan_principal(A, years, annual_rate), expected);
    }
}
